### ISSUE TITLE
Yolov8 pre-training  both weights and cfg

### ISSUE THREAD

**Issue by emre888** [ultralytics/ultralytics#1388]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Like previous yolo models  do "--weight  & --cfg ", when I want to train using pretrained model but different cfg layers, does yolov8 provide this

**Comment by Hxiaowu666** [comment]
I have the same problem. Have you solved it

**Comment by glenn-jocher** [comment]
👋 Hello @emilybarriere, 

I apologize if my response was unclear. However, I did not see a specific problem mentioned in your initial question. Could you please provide more details about the specific problem you are facing with YOLOv8? 

If you are referring to a previous question or issue, please mention the corresponding question or issue number and I would be happy to take a look.

**Comment by andrey101010** [comment]
I have a similar problem with this topic:

"Like previous yolo models do "--weight & --cfg ", when I want to train using pretrained model but different cfg layers, does yolov8 provide this"

Is there any solution?

**Comment by glenn-jocher** [comment]
@andrey101010 thank you for reaching out to us with your question.

In YOLOv8, you have the flexibility to use a pretrained model and customize the configuration (cfg) layers to suit your needs. You can load a pretrained model using the `--weights` option, and you can specify a different cfg file using the `--cfg` option.

By using the `--weights` option, you can load pre-trained weights from a previous training session or from a published model. This allows you to leverage the learned features and speed up the training process.

Similarly, the `--cfg` option allows you to specify a different configuration file. This gives you the ability to modify the architecture of the network by changing the layer configurations, such as the number of layers, filters, or activation functions.

By combining these options, you can train a YOLOv8 model using a pretrained model as a starting point and customize the network architecture to fit your specific requirements.

I hope this helps! Let me know if you have any further questions or need any additional assistance.

**Comment by andrey101010** [comment]
thx for the detailed answer. However for my case pretrained=True argument was the right choise. However, this website is reach in detail: https://docs.ultralytics.com/modes/train/#arguments

**Comment by glenn-jocher** [comment]
@andrey101010 thank you for your feedback! In your case, if you are looking to use a pretrained model directly without any modifications to the configuration (cfg) layers, the `pretrained=True` argument is indeed the right choice. This argument allows you to use the weights and architecture of a pretrained model without any changes.

I understand that the website you mentioned, which is https://docs.ultralytics.com/modes/train/#arguments, provides detailed information about the different arguments for training in YOLOv8. You can refer to this documentation for a thorough understanding of all the available options and their usage.

If you have any further questions or need additional assistance, please feel free to ask.

**Comment by RoeschFlo** [comment]
Hello Together :D
i have a existing ros software in which yolo_v3 was used. There I use:
# network files
		<param name="object_config_path" value="/data/no_ppt/yolov3-tiny.cfg" />
		<param name="object_weights_path" value="/data/no_ppt/yolov3-tiny.weights" />
		<param name="object_labels_path" value="/data/labels.txt" />
		<param name="object_engine_path" value="/data/no_ppt/tinyv3.engine" />
		<param name="object_network_type" value="yolov3-tiny" />

Is it possible to export all this files in the same format out of a existing yolov8 model? So that i have just chance these files and do some minor adjustments?
thank you

**Comment by glenn-jocher** [comment]
@RoeschFlo hello,

Yes, it is possible to replace your existing YOLOv3 files with YOLOv8 equivalents. You would follow a similar process - training your YOLOv8 model and then exporting the required files. 

After doing that, you would simply replace your existing files - you would update the `object_config_path` with your new YOLOv8 cfg file, the `object_weights_path` with your new weights file, and so on. 

If there are any changes in the label names or number of classes between YOLOv3 and YOLOv8, you would also need to update your labels file accordingly.

Just remember to confirm that any software or packages that you're using are compatible with YOLOv8. 

I hope this helps! Let me know if you have any further questions.

**Comment by kanis777** [comment]
hi @glenn-jocher .So when i run for train - i get a folder with weights containing best.pt and last.pt ryt. so y is it named weights .wt does the model actually contain- the weights like importance of particular classes or ?

**Comment by glenn-jocher** [comment]
Hello @kanis777,

Great question! When you train a YOLOv8 model, the resulting `best.pt` and `last.pt` files in the `weights` folder are indeed the model's weights. These files encapsulate the learned parameters of the model, which include the weights and biases of the neural network layers.

Here's a brief breakdown of what these files represent:
- **`best.pt`**: This file contains the weights of the model that achieved the best performance on the validation set during training.
- **`last.pt`**: This file contains the weights of the model from the final epoch of training.

These weights are crucial as they determine how the model processes input data to make predictions. They do not directly represent the importance of particular classes but rather the learned features and patterns from the training data that enable the model to perform object detection effectively.

If you need to export these weights to a different format for compatibility with your existing ROS setup, you can use the `export` function in YOLOv8. For example, to export to ONNX format, you can use:

```python
from ultralytics import YOLO

# Load the trained model
model = YOLO('path/to/best.pt')

# Export the model to ONNX format
model.export(format='onnx')
```

This will generate an ONNX file that you can integrate into your ROS application. For more details on exporting models, you can refer to the [Ultralytics documentation](https://docs.ultralytics.com/modes/export/).

I hope this clarifies your question! If you have any more queries or need further assistance, feel free to ask. 😊
